harvester_e2e_tests/apis/test_hosts.py

```diff
@@ -15,281 +15,172 @@
+def test_update_node(api_client):
     """
     Test the hosts are the nodes which make the cluster
     Covers:
         hosts-04-Edit Config Add description and other details
     """
-    assert resp.status_code == 200, 'Failed to list nodes: %s' % (resp.content)
-    host_data = resp.json()
-    first_node = host_data['data'][0]
-    original_annotations = copy.deepcopy(first_node['metadata']['annotations'])
-    first_node['metadata']['annotations'] = {
-        'field.cattle.io/description': 'for-test-update',
-        'harvesterhci.io/host-custom-name': 'for-test-update'
-    }
-    try:
-        resp = utils.poll_for_update_resource(
-            request, admin_session,
-            host_data['data'][0]['links']['update'],
-            first_node,
-            harvester_api_endpoints.get_node % (host_data['data'][0]['id']))
-        updated_host_data = resp.json()
-        assert updated_host_data['metadata']['annotations'].get(
-            'field.cattle.io/description') == 'for-test-update'
-        assert updated_host_data['metadata']['annotations'].get(
-            'harvesterhci.io/host-custom-name') == 'for-test-update'
-    finally:
-        # change it back to original
-        if updated_host_data:
-            updated_host_data['metadata']['annotations'] = original_annotations
-            utils.poll_for_update_resource(
-                request, admin_session,
-                updated_host_data['links']['update'],
-                updated_host_data,
-                harvester_api_endpoints.get_node % (updated_host_data['id']))
-
-
-@pytest.mark.hosts_p1
-@pytest.mark.p1
-def test_update_using_yaml(request, admin_session, harvester_api_endpoints):
+
+    fields = ('field.cattle.io/description', 'harvesterhci.io/host-custom-name')
+
+    _, nodes_info = api_client.hosts.get()
+
+    # There is no different to update any nodes, so we use the last.
+    node = nodes_info['data'][-1]
+    original_annotations = {k: node['metadata']['annotations'].get(k, "")
+                            for k in fields}
+
+    test_annotations = {k: f"Test_update_{k}_{datetime.now()}" for k in fields}
+    test_data = dict(metadata=dict(annotations=test_annotations))
+
+    status_code, node_stats = api_client.hosts.update(node['id'], test_data)
+
+    not_updated_fields = list()
+    for k, v in test_annotations.items():
+        if node_stats['metadata']['annotations'].get(k) != v:
+            not_updated_fields.append((k, v, node_stats['metadata']['annotations'].get(k)))
+
+    assert len(not_updated_fields) == 0, (
+        f"Fields not fully updated, API return `{status_code}`\n"
+        "\n".join(f"field:{k}, expected: {v}, got {o}" for k, v, o in not_updated_fields)
+    )
+
+    # For teardown
+    _ = api_client.hosts.update(node['id'],
+                                dict(metadata=dict(annotations=original_annotations)))
+
+
+@pytest.mark.dependency(depends=["get_host"])
+@pytest.mark.hosts
+@pytest.mark.p2
```

Comment:
As we have started following priority level starting P0 in Harvester project. So, P1 becomes P0 and P2 becomes P1. Could we update this?

Reply:
surely